modules/build/controllers/service/service.go

```diff
@@ -139,13 +140,11 @@ func (p populator) populateBuild(buildKey, namespace string, build *riov1.ImageB
 	if svc.Spec.Template {
 		return nil
 	}
-
-	rev := build.Revision
-	if rev == "" {
+	if build.Revision == "" {
 		return nil
 	}
 
-	trName := name.SafeConcatName(buildKey, name.Hex(build.Repo, 5), name.Hex(rev, 5))
+	trName := name.SafeConcatName(buildKey, name.Hex(build.Repo, 5), name.Hex(build.Revision, 5))
```

Comment:
note the taskrun name is still using revision, as does the rev passed into task git resource